nav2_planner/src/planner_server.cpp

```diff
@@ -735,14 +737,22 @@ void PlannerServer::exceptionWarning(
   const geometry_msgs::msg::PoseStamped & start,
   const geometry_msgs::msg::PoseStamped & goal,
   const std::string & planner_id,
-  const std::exception & ex)
+  const std::exception & ex,
+  std::string & error_msg)
 {
-  RCLCPP_WARN(
-    get_logger(), "%s plugin failed to plan from (%.2f, %.2f) to (%0.2f, %.2f): \"%s\"",
-    planner_id.c_str(),
-    start.pose.position.x, start.pose.position.y,
-    goal.pose.position.x, goal.pose.position.y,
-    ex.what());
+  error_msg = planner_id +
```

Comment:
Please reformat back in original style.

Reply:
Ok, Bring on c++20 for std::format.  

Do you have a preference for snprintf or sstringstream?

Reply:
I went with stringstream.